llvm/lib/Target/AMDGPU/GCNHazardRecognizer.cpp

```diff
@@ -3044,9 +3044,8 @@ bool GCNHazardRecognizer::fixVALUMaskWriteHazard(MachineInstr *MI) {
     return false;
 
   const Register HazardReg = HazardDef->getReg();
-  auto *HazardRegRC = TRI->getPhysRegBaseClass(HazardReg);
-  bool IsSGPR32 = (HazardRegRC == TRI->getSGPRClassForBitWidth(32)) ||
-                  HazardReg == AMDGPU::VCC_LO || HazardReg == AMDGPU::VCC_HI;
+  bool IsSGPR32 = (AMDGPU::SReg_32RegClass.contains(HazardReg) ||
+                   HazardReg == AMDGPU::VCC_LO || HazardReg == AMDGPU::VCC_HI);
```

Comment:
SReg_32 already covers vcc_lo/vcc_hi

Reply:
Thanks. My original method of accessing the RC didn't work for VCC.